galeracluster/source/kb/faq.rst

```diff
@@ -0,0 +1,299 @@
+
+.. note:: For more information, see :doc:`isolationlevels`.
+
+
+.. rubric:: How are DDL's Handled by Galera Cluster?
+.. _`faq-ddl-handled-galera`:
+
+For :abbr:`DDL (Data Definition Language)` statements and similar queries, Galera Cluster has two modes of execution:
+
+- :term:`Total Order Isolation`: A query is replicated in a statement before executing on the master.  The node waits for all preceding transactions to commit and then all nodes simultaneously execute the transaction in isolation.
+
+- :term:`Rolling Schema Upgrade`: Schema upgrades run locally, blocking only the node on which they are run.  The changes do not replicate to the rest of the cluster.
+
+.. note:: For more information, see :doc:`schemaupgrades`.
+
+
+.. rubric:: Is GCache a Binlog?
+.. _`faq-gcache-binlog`:
+
+The :term:`Write-set Cache`, which is also called GCache, is a memory allocator for write-sets.  Its primary purpose is to minimize the write-set footprint in RAM.  It is not a log of events, but rather a cache.
+
+- GCache is not persistent.
+- Not every entry in GCache is a write-set.
+- Not every write-set in GCache will be committed.
+- Write-sets in GCache are not allocated in commit order.
+- Write-sets are not an optimal entry for the binlog, since they contain extra information.
+
+Nevertheless, it is possible to construct a binlog out of the write-set cache.
+
+
+.. rubric:: Should the binary log be enabled with Galera?
+.. _`faq-gcache-binlog`:
+
+Standard MySQL replication uses the binary log for replicating. However, Galera doesn't use the binary log.  Nevertheless, there may be situations in which you might want to use point-in-time recovery methods to restore tables or data since the last backup.
+
+You might also want to attach an asynchronous slave to one of your nodes, using standard MySQL replication and set it on a delay.  This can also help with recovering tables and data lost since the last backup was made.
+
+
+-----------------------------------------------
+Questions about Potential Problems
+-----------------------------------------------
+
+.. rubric:: What typically Causes a Cluster to Stop?
+.. _`faq-what-causes-galera-to-stop`:
+
+Although it doesn't happen often, there are several reasons a Galera cluster might crash. Below is a list of them, grouped by type of cause:
+
+**Pysical Server & Related Causes**
+- The nodes are out of disk space;
+- The operating systems are swapping or have a high I/O Wait
+
+**Storage Engine Causes**
+- The InnoDB storage engine crashes;
+- Using MyISAM tables, which is still experimental;
+- Creating or dropping tables that don't have a primary key
+
+**Configuration Problems**
+- Incompatible Changes to Parameters in the MySQL Configuration File;
+- Setting binlog_format to only ROW, instead of MIXED

```

Comment:
This should maybe be `Setting binlog_format to MIXED instead of ROW.` Only ROW binlog format is fully supported.